product_code/lagom/user_service/api/src/main/scala/de/upb/cs/uc4/user/model/user/Student.scala

```diff
@@ -21,6 +22,57 @@ case class Student(username: String,
     copy(username.trim, role, address.trim, firstName.trim, lastName.trim,
       picture.trim, email.trim, birthDate.trim, immatriculationStatus.trim, matriculationId.trim)
   }
+
+  /** @inheritdoc */
+  override def validate: Seq[SimpleError] = {
+    val fos = List("Computer Science","Philosophy","Media Sciences", "Economics", "Mathematics", "Physics", "Chemistry",
+      "Education", "Sports Science", "Japanology", "Spanish Culture", "Pedagogy", "Business Informatics", "Linguistics")
+
+    var errors = super.validate.asInstanceOf[List[SimpleError]]
+    if(!(matriculationId forall Character.isDigit) || !(matriculationId.toInt > 0) || !(matriculationId.toInt < 10000000)) {
+      errors :+= SimpleError("matriculationId", "Matriculation ID must be a number between 1 and 9999999.")
+    }
+    if(!(semesterCount > 0)) {
+      errors :+= SimpleError("semesterCount", "Semester count must be a positive integer.")
+    }
+    if(!(fieldsOfStudy forall fos.contains)) {
+      errors :+= SimpleError("fieldsOfStudy", "Fields of Study must be one of [..].")
+    }
+    errors
+  }
+
+
+  /** 
+    * Compares the object against the user parameter to find out if fields, which should only be changed by users with elevated privileges, are different.
+    * Returns a list of SimpleErrors[[de.upb.cs.uc4.shared.client.SimpleError]]
+    * 
+    * @param user 
+    * @return Filled Sequence of [[de.upb.cs.uc4.shared.client.SimpleError]]
+    */
+  override def checkEditableFields (user: User): Seq[SimpleError] = {
+    if(user.isInstanceOf[Student]){
+      throw new Exception("Tried to parse a non-Student as Student.")
+    }
+    val student = user.asInstanceOf[Student]
```

Comment:
You check if the user object is a `Student`, but the exception shall be thrown when the object is not a `Student`.